app/elements/pw-shell.html

```diff
@@ -250,13 +286,44 @@
 
       // Save the initial URL so that we don't double-render the first load.
       this._currentPageURL = this._sectionRoute.path;
+
+      // Save which version of docs we're looking at, since we need to
+      // maintain this state across the app. Note that not all sections are versioned.
+      if (this._sectionData.version === '1.0' || this._sectionData.version === '2.0')
+        this.activeDocsVersion = this._sectionData.version;
+    },
+
+    _listboxVersionChanged: function(event) {
+      var newSelection = event.detail.item.textContent;
+      if (newSelection === this.activeDocsVersion)
+        return;
+      this.activeDocsVersion = newSelection;
+      this.set('_sectionRoute.path', '/' + this.activeDocsVersion + '/start/');
     },
 
     _toggleDrawer: function() {
+      // Sync the version selector with the active docs version.
+      this.$.versionSelect.selected = this.activeDocsVersion == '1.0' ? 0 : 1;
+
+      var links = this.querySelectorAll('a[data-version]');
```

Comment:
For versioned pages (e.g. "/1.0/start/"), we only send v1 links (remember that even the drawer links on mobile are light DOM children of `<pw-shell>` that are _projected_ into the drawer through `<content select=".site-nav">`).

For non-versioned pages (e.g. "/blog/"), we could consider only including 2.0 links (forcing docs users to 2.0 even if they previously  browsed 1.0), but that's probably a little harsh. Another less expensive way to do this toggling (instead of iterating through all the links) is to make `<pw-shell>.activeDocsVersion` reflect to attribute, then style the other links in _main.scss, like:

```css
pw-shell[active-docs-version=1.0] a[data-version=2.0] {
  display: none;
}
```